jax/_src/cudnn/scaled_matmul_stablehlo.py

```diff
@@ -492,13 +492,17 @@ def quantize(x, config):
   elif config.mode == "nvfp4":
     assert config.scale_type == jnp.float8_e4m3fn
     assert config.global_scale.dtype == jnp.float32
+    SCALE_MAX = jnp.finfo(config.scale_type).max.astype(x.dtype)
 
-    scales = scales / config.global_scale
-    scales_q = jax.lax.optimization_barrier(scales.astype(jnp.float8_e4m3fn))
-    scaled_x = x / (scales_q.astype(jnp.float32) *
-                    config.global_scale).astype(x.dtype)
+    prev_amax = config.global_scale * (MAX * SCALE_MAX)
+    scales_q = jnp.clip(
+        (amax / prev_amax) * SCALE_MAX, 0, SCALE_MAX
+    )
+    scaled_x = x / scales_q
+    scales_q = scales_q.astype(config.scale_type)
```

Comment:
I don't think we should change the order and we should use the fp8 scales during the scaling of the x, meaning it has to be scales->fp8->cvt->fp32 for the scales.